Fushigi/ui/widgets/CourseScene.cs

```diff
@@ -2483,6 +2635,51 @@ private void SelectActorToAddLayer()
                 }
             }
 
+            // TODO, maybe find a way to combine with SelectActorToAddLayer(), if that's needed
+            private void SelectLayerToAdd()
+            {
+                ImGui.InputText("Search", ref mAddLayerSearchQuery, 256);
+
+                string[] Layers = ["PlayArea", "DecoArea", "DecoAreaFront"];
+                Layers = Layers.Concat(DistantViewManager.ParamTable.Layers.Keys)
+                    .Except(mLayersVisibility.Keys)
+                    .ToArray();
+                var fileteredLayers = Layers.ToImmutableList();
+                
+                if (mAddLayerSearchQuery != "")
+                {
+                    fileteredLayers = FuzzySharp.Process.ExtractAll(mAddLayerSearchQuery, Layers, cutoff: 65)
+                        .OrderByDescending(result => result.Score)
+                        .Select(result => result.Value)
+                        .ToImmutableList();
+                }
+
+                if (ImGui.BeginListBox("Select the layer you want to add the actor to.", ImGui.GetContentRegionAvail()))
+                {
+                    for (var i = 0; i < fileteredLayers.Count; i++)
+                    {
+                        var layer = i < 2 ? 
+                            fileteredLayers[i]+$" ({mLayersVisibility.Count(x => Regex.Replace(x.Key, @"\\d+", "").Contains(fileteredLayers[i]))}/10)":
+                            fileteredLayers[i];
+
+                        if(layer.Contains("(10/10)"))
```

Comment:
While this code may be very easy to read it's also very error prone
there should be a local variable `layerCount` and a (local) `const int MaxLayerCount = 10;`

Also `ImGui.BeginDisabled` has a condition parameter which I encourage using instead of putting both Begin and End in `if` clauses